app/main/movie_service.py

```diff
@@ -0,0 +1,39 @@
+import os
+import datetime as dt
+import requests
+
+from app import db
+from app.main.models import Movie
+
+# is it ok to import database inside this module?
+
+MOVIE_DB_IMAGE_URL = "https://image.tmdb.org/t/p/w500"
+MOVIE_API_URL = "https://api.themoviedb.org/3"
+
+
+class MovieService:
+    def __init__(self, headers):
+        self.headers = headers
+
+    def get_movie(self, movie_id):
+        self.url = os.path.join(MOVIE_API_URL, "movie", f"{movie_id}?language=en-US")
+        response = requests.get(self.url, headers=self.headers).json()
```

Comment:
What in case of errors from external service? You shouldn't trust anything what isn't yours.

Reply:
TMDb api returns empty list when no results are found, e.g. when invalid title is put. I catch this inside select.html and generate error message when it occurs. My initial thought was to use this functionality to catch all errors but I don't know if that's the best approach. What do you think? 
here is sample code: 
    def get_movies_to_select(self, title):
        url = os.path.join(MOVIE_API_URL, "search", "movie")
        params = {"query": f"{title}"}
        response = requests.get(url, headers=self.headers, params=params)
        if response.status_code != 200:
            data = []
        else:
            response = response.json()
            data = response["results"]
        return data

and in select.html: 
{% if data == [] %}

  <h1 class="heading">Error</h1>

      <h2>Unfortunatelly we could not find this movie in the TMDb database.</h2>
      <p>Please click <a href="{{ url_for('main.home') }}">here</a>  to come back to the main page and try different movie. </p>

{% else %}

Reply:
I think it is good enough for now